### PR TITLE
refactor: fixed ESLint warnings

### LINKED ISSUE #44: [ESLint] Fix the warnings.
Screenshot 2024-09-20 at 11 43 02 AM

### PR DESCRIPTION
**Pull request by byhlel**
Resolves #44

### CODE REVIEW

File: src/miningRulesCore/preProcessing.js [hunk 1]
```diff
@@ -80,20 +84,22 @@ const cleanGroupings = (xmlFile, groupingMetaData) => {
 const addToGroupings = (xmlFile, groupingMetaData) => {
     const fileID = xmlFile.filePath;
     for (const groupingCategoryID in groupingCategories) {
```
Comment (SaharMehrpour) [inline src/miningRulesCore/preProcessing.js:83]: What is the change in this block?

Comment (SaharMehrpour) [reply src/miningRulesCore/preProcessing.js:83]:
```suggestion
    for (const groupingCategoryID in groupingCategories) {
        if (!groupingCategories.hasOwnProperty(groupingCategoryID)) continue;
```

File: src/miningRulesCore/preProcessing.js [hunk 2]
```diff
@@ -21,6 +21,10 @@ export const createGroupingMetaData = () => {
     const groupMapping = {};
     const fileMapping = {};
     for (const key in groupingCategories) {
+        if (groupingCategories.hasOwnProperty(key)) {
```
Comment (SaharMehrpour) [inline src/miningRulesCore/preProcessing.js]:
```suggestion
        if (!groupingCategories.hasOwnProperty(key)) continue;
```

**Review by SaharMehrpour — CHANGES_REQUESTED** [review]

**Comment by byhlel** [comment]
The suggested way of refactoring has been implemented!

**Review by SaharMehrpour — APPROVED** [review]

**Pull request merged**